pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -183,6 +194,8 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pr *v1beta1.PipelineRun)
 		logger.Errorf("Failed to fetch pipeline func for pipeline %s: %w", pr.Spec.PipelineRef.Name, err)
 		pr.Status.MarkFailed(ReasonCouldntGetPipeline, "Error retrieving pipeline for pipelinerun %s/%s: %s",
 			pr.Namespace, pr.Name, err)
+		span.SetStatus(codes.Error, "Failed to fetch pipelinefunc")
+		span.RecordError(err)
```

Comment:
Shall we record an error in all failure cases, or is there a specific reason to capture it only in case of failing to fetch the pipeline spec?

Reply:
we should record error wherever it makes sense to be able to debug issues by visualizing in jaeger. I just added one as an example in this PR.